source/Nevermore/Mapping/KeyAllocator.cs

```diff
@@ -28,15 +31,21 @@ public int NextId(string tableName)
             return allocation.Next();
         }
 
+        public async Task<int> NextIdAsync(string tableName, CancellationToken cancellationToken)
```

Comment:
Given that this holds blocks, 19 out of 20 times it's going to return synchronously. As such, might be nice to return ValueTask (and plumb that back up the stack through AllocateNextAsync) to communicate that

Reply:
Yeah that occurred to me as I wrote this, I opted for `Task` anyway due to the "consume at most once" restriction on `ValueTask` feeling a bit dangerous but you're right it does seem like a good fit. I'm still not sure how to evaluate that.

One point in favour of choosing `ValueTask` here is that it's relatively safe to later on go `ValueTask` -> `Task` (might break at compile time on `AsTask()`, but it won't break anything at runtime at least) but not safe to go the other way. 🤔

Reply:
I decided to go with `ValueTask` in the end 👍